WMF Framework/Remote Notifications/RemoteNotificationsAPIController.swift

```diff
@@ -116,25 +118,33 @@ struct RemoteNotificationsAPIController {
 
     public func markAsRead(_ notifications: Set<RemoteNotification>, completion: @escaping (Error?) -> Void) {
         let maxNumberOfNotificationsPerRequest = 50
+        let notifications = Array(notifications)
+        let split = notifications.chunked(into: maxNumberOfNotificationsPerRequest)
 
-        guard notifications.count <= maxNumberOfNotificationsPerRequest else {
-            // TODO: Split requests? 50 is the limit.
-            assertionFailure()
-            return
-        }
-        
-        
-
-        request(Query.markAsRead(notifications: notifications), method: .post) { (result: MarkReadResult?, _, _, error) in
-            if let error = error {
-                completion(error)
+        split.asyncMap({ (notifications, completion: @escaping (Error) -> Void) in
+            request(Query.markAsRead(notifications: notifications), method: .post) { (result: MarkReadResult?, _, _, error) in
+                if let error = error {
+                    completion(error)
+                }
+                guard let result = result else {
+                    assertionFailure("Expected result; make sure MarkReadResult maps the expected result correctly")
+                    completion(MarkReadError.noResult)
+                    return
+                }
+                if let error = result.error {
+                    completion(error)
+                }
+                if !result.succeeded {
+                    completion(MarkReadError.unknown)
+                }
```

Comment:
This is missing a completion for the non-error case. Could implement `asyncCompactMap` and add `completion(nil)`